gui/app/periodic_graph_painter.h

```diff
@@ -0,0 +1,44 @@
+#ifndef CIRCULARLINESERIESPAINTER_H
+#define CIRCULARLINESERIESPAINTER_H
+
+#include "qnanoquickitem.h"
+#include "qnanoquickitempainter.h"
+#include <QQuickItem>
+#include <qnanocolor.h>
+
+class PeriodicGraphPainter : public QNanoQuickItemPainter {
+public:
+  PeriodicGraphPainter();
+
+  void paint(QNanoPainter *p);
+
+  void synchronize(QNanoQuickItem *item);
+
+private:
+  QVector<QPointF> dataset_;
+
+  float max_value_ = 100;
+  float min_value_ = 0;
+  float period_in_sec = 30;
+  QNanoColor color_fill_ = QNanoColor(255, 255, 255, 90);
+  QNanoColor color_gradient_end_ = QNanoColor(255, 255, 255, 90);
+  QNanoColor color_line_ = QNanoColor(255, 255, 255, 255);
+  QNanoLinearGradient bg;
+  QNanoColor baseline_color_ = QNanoColor("#13345B");
+  bool show_baseline_ = true;
+
+  float baseline_value_ = 0;
+
+  float calculateRealX(float timeX) {
+    float result = width() - (((-1 * timeX) / period_in_sec) * width());
```

Comment:
* This is simply width() * (1.0 + timeX / period_in_sec), right?
* OK I understand what "period" means now - it's just the range of the graph. I would suggest to rename the stuff from PeriodicGraph to simply TimeSeriesGraph.